### PR TITLE
Android annotations: copy existing classes & Kotlinify manager classes

### LINKED ISSUE #1492: Android Annotations: copy existing classes & Kotlinify manager classes
[Android annotations API proposal](https://github.com/maplibre/maplibre-native/blob/main/design-proposals/2023-06-17-android-annotations.md) implementation issue as a part of https://github.com/maplibre/maplibre-native/issues/1491

---

First, it makes sense to copy the existing classes into `maplibre-native`, to fix imports, etc., so that we can then continue with modifications. In the same turn, we can Kotlinify the manager classes, as they will not require much attention overall per the proposal.

* Copy existing classes from maplibre annotations plugin
* Resolve problems
* Kotlinify manager classes

### PR DESCRIPTION
**Pull request by fynngodau**
[Android annotations API proposal](https://github.com/maplibre/maplibre-native/blob/main/design-proposals/2023-06-17-android-annotations.md) implementation PR as a part of https://github.com/maplibre/maplibre-native/issues/1491. Closes #1492.

---

* Copy existing classes from maplibre annotations plugin
* Remove "This file is generated" comments
* Rename `Annotation` to `AbstractAnnotation` due to naming conflict (pending re-evaluation)
* Fix imports
* Kotlinify Manager classes
* Copy tests for Manager classes, and Kotlinify them as well. The auto-generated code was prettified by me; it could be formatted better, but for these boilerplate tests it seems okay.

### CODE REVIEW

**Comment by louwers** [comment]
I will update CI so that the checks run.

File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/FillManager.kt [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+     *
+     * Fills are going to be created only for features with a matching geometry.
+     *
+     *
+     * All supported properties are:<br></br>
+     * FillOptions.PROPERTY_FILL_OPACITY - Float<br></br>
+     * FillOptions.PROPERTY_FILL_COLOR - String<br></br>
+     * FillOptions.PROPERTY_FILL_OUTLINE_COLOR - String<br></br>
+     * FillOptions.PROPERTY_FILL_PATTERN - String<br></br>
+     * Learn more about above properties in the [Style specification](https://www.mapbox.com/mapbox-gl-js/style-spec/).
```
Comment (louwers) [inline platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/FillManager.kt:135]: Let's replace there while we are at it
```suggestion
     * Learn more about above properties in the [Style specification](https://maplibre.org/maplibre-style-spec/).
```

File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/FillManager.kt [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
```
Comment (louwers) [inline platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/FillManager.kt:111]:
```suggestion
     * Learn more about above properties in the [Style specification](https://maplibre.org/maplibre-style-spec/).
```

File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/CircleManager.kt [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+     *
+     * All supported properties are:<br></br>
+     * CircleOptions.PROPERTY_CIRCLE_RADIUS - Float<br></br>
+     * CircleOptions.PROPERTY_CIRCLE_COLOR - String<br></br>
+     * CircleOptions.PROPERTY_CIRCLE_BLUR - Float<br></br>
+     * CircleOptions.PROPERTY_CIRCLE_OPACITY - Float<br></br>
+     * CircleOptions.PROPERTY_CIRCLE_STROKE_WIDTH - Float<br></br>
+     * CircleOptions.PROPERTY_CIRCLE_STROKE_COLOR - String<br></br>
+     * CircleOptions.PROPERTY_CIRCLE_STROKE_OPACITY - Float<br></br>
+     * Learn more about above properties in the [Style specification](https://www.mapbox.com/mapbox-gl-js/style-spec/).
```
Comment (louwers) [inline platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/CircleManager.kt:135]:
```suggestion
     * Learn more about above properties in the [Style specification](https://maplibre.org/maplibre-style-spec/).
```

File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/CircleManager.kt [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
```
Comment (louwers) [inline platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/CircleManager.kt:162]:
```suggestion
     * Learn more about above properties in the [Style specification](https://maplibre.org/maplibre-style-spec/).
```

File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/DraggableAnnotationController.java [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+@UiThread
+final class DraggableAnnotationController {
+
+  private static DraggableAnnotationController INSTANCE = null;
+
+  public static DraggableAnnotationController getInstance(MapView mapView, MapLibreMap mapboxMap) {
```
Comment (louwers) [inline platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/DraggableAnnotationController.java:29]: Let's rename this one.

File: platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/LineManager.kt [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+     * All supported properties are:<br></br>
+     * LineOptions.PROPERTY_LINE_JOIN - String<br></br>
+     * LineOptions.PROPERTY_LINE_OPACITY - Float<br></br>
+     * LineOptions.PROPERTY_LINE_COLOR - String<br></br>
+     * LineOptions.PROPERTY_LINE_WIDTH - Float<br></br>
+     * LineOptions.PROPERTY_LINE_GAP_WIDTH - Float<br></br>
+     * LineOptions.PROPERTY_LINE_OFFSET - Float<br></br>
+     * LineOptions.PROPERTY_LINE_BLUR - Float<br></br>
+     * LineOptions.PROPERTY_LINE_PATTERN - String<br></br>
+     * Learn more about above properties in the [Style specification](https://www.mapbox.com/mapbox-gl-js/style-spec/).
```
Comment (louwers) [inline platform/android/MapboxGLAndroidSDK/src/main/java/org/maplibre/android/annotations/LineManager.kt:169]:
```suggestion
     * Learn more about above properties in the [Style specification](https://maplibre.org/maplibre-style-spec/).
```

**Review by louwers — APPROVED** [review]
Feel free to merge once the comments have been addressed and CI passes. 👍 

I haven't tested it myself but I will do so once we merge this branch into `main`.

I merged `main` into `android-annotations` so I could see the diff without noise.

File: platform/android/MapboxGLAndroidSDK/src/test/java/org/maplibre/android/annotations/FillManagerTest.kt [hunk 7]
```diff
@@ -0,0 +1,739 @@
...
+import org.mockito.ArgumentCaptor
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito
+
+class FillManagerTest {
+    private val draggableAnnotationController = Mockito.mock(
+        DraggableAnnotationController::class.java
+    )
+    private val mapView: MapView = Mockito.mock(MapView::class.java)
+    private val mapboxMap: MapLibreMap = Mockito.mock(MapLibreMap::class.java)
```
[the remark this answers is not in the document]
Comment (fynngodau) [reply platform/android/MapboxGLAndroidSDK/src/test/java/org/maplibre/android/annotations/FillManagerTest.kt:35]: Definitely!

**Pull request merged**